symbolic-symcache/src/new/writer.rs

```diff
@@ -308,6 +313,113 @@ impl SymCacheConverter {
+                    for transformer in &self.transformers.0 {
+                        function = transformer.transform_function(function);
+                    }
+
+                    let string_bytes = &mut self.string_bytes;
+                    let strings = &mut self.strings;
+                    let name_offset = Self::insert_string(string_bytes, strings, &function.name);
+
+                    let (fun_idx, _) = self.functions.insert_full(raw::Function {
+                        name_offset,
+                        comp_dir_offset: u32::MAX,
+                        entry_pc: address,
+                        lang: Language::CSharp as u32,
+                    });
+                    fun_idx
+                };
+            }
+
+            let managed_dir = if record.managed_file_info.dir_str().is_empty() {
+                None
+            } else {
+                Some(record.managed_file_info.dir_str())
+            };
+            let mut location = transform::SourceLocation {
+                file: transform::File {
+                    name: record.managed_file_info.name_str(),
+                    directory: managed_dir,
+                    comp_dir: None,
+                },
+                line: record.managed_line,
+            };
+            for transformer in &self.transformers.0 {
+                location = transformer.transform_source_location(location);
+            }
+
+            let string_bytes = &mut self.string_bytes;
+            let strings = &mut self.strings;
+            let path_name_offset = Self::insert_string(string_bytes, strings, &location.file.name);
+            let directory_offset = location
+                .file
+                .directory
+                .map_or(u32::MAX, |d| Self::insert_string(string_bytes, strings, &d));
+
+            let (file_idx, _) = self.files.insert_full(raw::File {
+                path_name_offset,
+                directory_offset,
+                comp_dir_offset: u32::MAX,
+            });
+
+            let source_location = raw::SourceLocation {
+                file_idx: file_idx as u32,
+                line: location.line,
+                function_idx: function_idx as u32,
+                inlined_into_idx: u32::MAX,
+            };
+
+            match self.ranges.entry(address) {
+                btree_map::Entry::Vacant(entry) => {
+                    entry.insert(source_location);
+                }
+                btree_map::Entry::Occupied(mut entry) => {
+                    // TODO: This exists in native-only mappings, but we don't know yet if it's
+                    // possible to generate these types of records in managed code.
+                    // println!(
+                    //     "Found what's probably an inlined source {}::{}:L{}",
+                    //     record.managed_file_info.path_str(),
+                    //     record.managed_symbol,
+                    //     record.managed_line,
+                    // );
```

Comment:
i'll revisit this once i unconfuse myself about the logging situation in this crate

Reply:
well, the logging situation is that we don’t have any logging 🤷🏻‍♂️

Reply:
oh okay, i wasn't sure if that was the case, or if it was just that i couldn't figure out which crate we were using due to a lack of familiarity with the ecosystem. i'll do a follow-up that adds in some logging, although in retrospect i think it may be better for me to figure out how to send these as sentry events instead 🤔

Reply:
sentry's sdk doesn't really have a zero-overhead-if-unused compilation behaviour that makes it suitable for use in a library though

Reply:
Well, `sentry-core` without the `client` feature is pretty close to "zero-overhead-if-unused". Though we don’t really have a good precedent for using that. On the other hand though, this sounds like a good incentive to actually start making use of that.

Reply:
it looks like we're mostly okay (?) with adding some degree of logging, so i'll play around with the sentry sdk and its settings to see what i can get up and running. this'll be in a different PR though, and maybe we'll be happy enough with a different approach that we'll decide to ditch this? we'll see.

Reply:
I'm still very suspicious of using sentry in a library.  If you need to report errors why not use established `log::error!` or `tracing::error` which gives a little more fidelity, sentry can hook into the logging frameworks to capture those.

However if we really have errors we need to be very careful about the decision to ignore them.  Libraries really want to give this choice to the users instead of jumping to reporting measures themselves.  So this entire choice thing seems like a code smell to me.  IIRC we had a discussion about what to do with errors as we were not sure about our interpretations of what it **should** be like, and the conclusion then was that **for now** we need to be very strict so that it blows up in our face if our assumptions are wrong.  And **later**, when this moves out of being experimental, we should make it more lenient again.  It seems to me this is all out this later part already so somewhat jumping the gun.